packages/css/src/plugins/__tests__/increase-specificity.test.ts

```diff
@@ -0,0 +1,55 @@
+import postcss from 'postcss';
+
+import { increaseSpecificity } from '../increase-specificity';
+
+const transform = (css: TemplateStringsArray) => {
+  const result = postcss([increaseSpecificity()]).process(css[0], {
+    from: undefined,
+  });
+
+  return result.css;
+};
+
+describe('increase specicifity plugin', () => {
```

Comment:
Worth a test for some pseudos, eg. this is expected?
```css
@media (…) {
  .abc123:hover:not(#\9) {
    color: red;
  }
}
```

What about keyframes, supports, other at-rules?  No conerns?

Reply:
I'll add some tests to cover them (specifically: they shouldn't directly be affected, only the class names!).

Reply:
86bd8eb5d7c548a0a4a8d5630bcc246a8eb0e9ac was there an issue 🤔 I figured this could go first.

How about a case where `selector = '.abc123 button'` or something nested?  Anything like that?

Reply:
Yeah there was! Before/after weren't applied if the `has:` was at the end. I'll do more edge cases exploration tests.

Reply:
since kylor already mentioned nesting (`.abc123 { button { ... } }`), I'll also mention that some tests for funky cases involving `&` would be good too

examples:

```css
div & { /* ... */ }
div:hover & { /* ... */ }
div &:hover { /* ... */ }
```

Reply:
All of those will still have a class name prefixing it post atomic-ification. Will add them though! Push soon.